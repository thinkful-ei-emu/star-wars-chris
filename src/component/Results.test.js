// make React available
import React from 'react';

//make the ReactDOM available, necessary for rendering the component
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

//make the App component available
import Results from './Results';

describe('SearchBar component', () => {
//this is the test case
    it('renders without crashing', () => {
        // first create a DOM element to render the component into
        const div = document.createElement('div');

        //render the component, this is the actual test, if something is wrong it will fail here
        ReactDOM.render(<Results results={[{name:'Your'},{name:'Results'},{name:'Here'}]}  />, div);

        //clean up code
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<Results results={[{name:'Your'},{name:'Results'},{name:'Here'}]}  />)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
});