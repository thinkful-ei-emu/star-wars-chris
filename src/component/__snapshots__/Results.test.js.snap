// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders the UI as expected 1`] = `
<div>
  <h2>
    Results:
  </h2>
  <ul>
    <li>
      Your
    </li>
    <li>
      Results
    </li>
    <li>
      Here
    </li>
  </ul>
</div>
`;
